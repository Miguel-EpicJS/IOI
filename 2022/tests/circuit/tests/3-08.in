385045b28fbc1903eca997d513040235e0229e01
1000 1000 5
-1 0 0 2 2 2 0 0 7 5 8 0 11 7 5 13 8 13 8 16 14 15 13 18 0 21 13 26 22 0 16 17 1 22 26 31 31 27 27 31 7 5 34 15 11 9 12 22 8 25 8 31 29 29 48 50 29 47 29 43 51 36 61 6 15 58 64 63 29 68 58 21 43 7 72 63 33 70 12 18 72 15 56 79 70 56 65 73 77 64 64 56 73 77 64 22 88 40 85 77 73 80 63 89 3 67 53 66 84 58 88 66 79 73 70 63 77 115 73 53 44 88 83 118 102 53 91 100 109 118 78 73 119 128 11 85 91 134 74 113 125 84 134 126 114 111 139 140 131 137 81 126 126 73 100 100 41 153 140 103 82 123 116 56 58 144 48 137 115 148 58 147 78 85 139 174 48 159 52 82 146 159 125 137 174 126 176 160 92 179 163 22 11 173 146 137 189 154 147 156 151 29 21 200 152 65 177 194 166 187 192 196 190 139 53 175 85 79 137 214 80 114 176 56 11 66 162 214 153 191 138 153 231 210 160 98 205 10 212 163 231 190 222 149 228 243 218 243 94 160 213 235 68 154 236 87 77 174 251 233 161 243 147 223 146 251 118 173 242 151 142 245 253 160 268 53 173 273 113 215 85 116 91 276 125 222 139 119 214 114 239 251 282 113 272 158 224 11 275 238 185 280 223 241 274 28 187 303 264 191 309 307 223 304 185 222 139 190 274 288 37 91 297 296 190 223 302 321 285 106 301 280 292 221 273 241 184 206 267 5 315 230 48 153 114 313 306 160 124 313 229 144 226 249 282 327 266 195 226 140 304 276 239 286 325 362 274 361 70 221 362 290 222 144 293 239 292 95 326 297 70 283 321 178 239 293 98 298 382 283 235 346 316 372 304 322 374 346 54 361 387 192 81 362 297 231 182 195 257 210 191 264 338 293 331 413 302 273 278 375 382 362 404 310 77 279 43 413 359 327 278 375 234 370 392 357 418 368 340 235 353 144 88 298 306 278 73 297 317 329 248 319 423 286 176 428 284 229 408 248 356 278 321 224 164 429 282 254 400 420 158 388 393 431 440 447 280 381 327 459 433 463 424 282 482 426 409 407 262 77 399 280 462 406 473 471 386 397 484 374 484 285 435 321 434 184 446 316 159 503 358 368 249 454 471 372 515 306 484 509 343 375 112 501 423 479 282 464 129 439 491 249 404 319 417 517 321 276 446 508 496 512 526 426 505 515 411 494 382 397 327 470 195 424 536 523 461 286 429 508 463 475 372 431 543 522 530 540 404 457 344 358 546 194 537 491 268 548 447 446 489 569 268 515 532 493 489 477 545 545 545 562 549 492 501 580 586 589 424 447 169 545 469 446 594 471 588 415 472 510 241 579 424 431 404 610 568 426 580 579 531 600 477 579 566 598 605 590 529 473 346 404 188 579 224 404 579 598 352 548 335 634 585 625 637 620 644 489 483 620 515 336 375 365 530 436 145 537 652 625 417 463 644 418 498 572 344 550 656 659 661 595 334 229 593 494 352 397 580 678 431 414 576 266 577 675 565 675 491 625 652 583 631 625 666 300 549 590 253 627 623 610 293 439 492 631 688 701 426 672 653 644 565 707 625 661 667 545 637 620 343 604 582 447 602 631 600 380 720 590 662 191 675 613 491 632 697 705 137 706 731 523 701 702 724 743 578 637 663 643 223 593 658 568 550 599 665 480 201 501 671 758 678 493 743 532 675 667 656 663 655 740 704 455 755 733 762 743 678 754 779 742 670 771 776 576 553 731 752 718 684 758 711 463 570 481 737 791 730 788 738 624 769 772 701 729 637 775 637 754 595 780 804 799 780 729 570 778 577 777 705 792 772 778 823 698 657 536 702 828 791 743 792 631 809 546 833 158 482 615 717 515 746 238 818 838 731 828 758 482 791 222 844 715 758 853 676 850 837 776 785 620 563 820 831 828 653 810 667 780 598 865 841 841 843 869 723 264 826 876 871 870 839 849 865 881 777 886 861 688 868 802 885 827 787 885 845 851 848 887 870 893 885 893 850 712 874 889 792 877 908 909 904 873 899 848 900 876 915 868 777 895 902 901 848 841 870 926 812 922 926 927 922 884 817 881 927 852 923 934 926 930 854 900 941 941 933 946 917 943 717 931 943 931 949 950 610 913 940 956 797 916 931 944 933 956 942 943 921 758 967 967 945 971 950 953 964 956 851 906 978 935 967 968 962 966 758 973 909 986 988 988 948 991 992 982 981 941 997 998 3 5 4 3 8 0 11 6 9 17 16 17 16 5 24 25 28 9 29 10 10 17 22 36 51 43 21 20 32 30 42 16 42 44 51 51 14 34 58 19 59 12 41 61 35 73 73 70 75 45 33 80 15 26 29 55 65 34 70 70 23 4 71 106 106 99 52 38 104 90 72 108 91 21 49 26 114 76 96 75 33 69 81 43 78 110 54 39 105 60 19 86 133 110 93 118 139 114 29 104 132 131 152 130 153 136 106 7 113 135 122 147 84 97 99 120 160 139 167 127 189 86 154 179 134 153 62 107 159 167 160 176 152 197 164 202 141 163 109 191 219 175 194 214 214 168 179 168 72 217 121 225 157 172 150 201 118 184 180 216 14 241 145 170 209 240 155 152 223 188 46 220 232 232 255 182 98 237 186 222 236 223 117 202 211 133 235 10 222 193 75 269 165 223 187 257 249 259 282 217 250 246 228 291 174 270 295 57 247 258 277 271 261 308 318 159 317 317 276 238 323 308 282 74 287 301 153 171 181 158 256 288 305 281 348 283 351 316 277 193 308 334 354 237 349 289 342 215 357 167 276 268 249 303 332 206 265 282 311 291 363 243 366 206 319 343 314 249 383 223 375 347 350 168 148 340 345 358 362 376 381 356 387 334 369 278 233 294 304 345 407 373 308 380 360 303 322 397 316 211 298 145 398 343 408 323 418 267 275 422 437 420 166 168 244 431 337 444 407 445 321 310 207 439 359 330 37 392 350 408 419 144 345 442 421 381 411 286 333 307 279 299 396 448 392 463 456 347 346 364 278 461 308 389 442 347 369 290 390 310 468 455 208 478 343 497 413 404 331 443 411 464 473 460 293 371 388 387 319 339 403 203 465 513 512 410 491 449 485 474 412 419 484 327 162 502 516 500 496 318 423 512 425 530 527 452 441 228 489 467 453 252 227 328 543 536 540 535 143 443 493 411 384 543 521 499 275 402 497 518 416 483 316 537 553 394 498 260 533 263 432 498 507 545 542 567 356 551 482 552 511 476 341 539 249 463 496 438 540 540 443 544 484 564 297 574 198 413 424 556 312 324 583 516 377 555 466 547 430 391 532 427 448 290 598 451 557 587 319 577 561 482 367 504 574 547 611 222 518 586 608 405 437 612 514 439 615 379 533 618 450 378 603 575 525 542 629 524 584 541 610 210 628 529 601 537 397 199 600 573 455 559 552 385 616 567 581 239 458 183 631 652 647 355 562 422 601 431 669 503 622 629 470 654 661 619 592 173 554 633 361 495 636 534 629 650 596 644 635 573 666 656 318 529 455 488 653 534 664 536 372 629 540 689 638 688 617 651 528 591 635 694 666 395 657 661 540 697 641 702 571 560 685 698 703 536 701 640 701 708 562 699 698 410 652 585 607 401 679 101 558 552 671 705 663 725 704 731 540 506 441 690 709 686 692 700 625 691 680 694 744 544 690 660 749 721 657 747 558 722 204 687 320 727 683 669 490 702 763 764 486 587 648 646 739 738 728 732 770 768 766 674 723 755 775 719 776 717 713 782 696 645 719 166 757 630 659 506 40 734 748 702 784 239 791 649 740 745 776 642 718 781 772 520 736 678 610 736 756 719 805 775 613 639 789 798 767 794 770 809 763 796 742 382 698 621 456 761 782 787 808 783 800 773 787 788 487 726 816 636 628 714 651 673 578 714 741 727 785 805 797 569 765 370 811 681 750 753 815 606 812 614 793 818 806 751 632 807 795 838 801 760 717 677 585 739 639 695 735 832 682 532 585 819 852 814 613 794 532 835 831 854 848 597 664 698 791 836 755 609 857 861 840 853 788 372 767 813 856 824 852 790 658 812 763 872 821 762 865 859 723 844 855 846 858 803 766 867 834 879 855 786 754 519 754 852 875 868 881 860 892 877 847 757 768 761 898 866 789 887 878 715 842 863 825 904 895 886 893 774 583 895 872 864 884 820 913 893 898 891 911 916 845 893 829 830 626 897 804 884 710 915 907 908 759 668 896 916 716 822 538 871 900 902 912 929 928 926 932 930 841 924 781 905 928 882 693 883 930 925 900 909 945 918 902 904 932 910 951 916 951 888 899 955 920 936 946 949 890 940 923 959 916 935 957 848 960 894 938 952 961 955 939 862 777 958 969 880 937 902 921 948 971 931 965 971 857 947 962 919 976 934 977 843 972 957 945 973 944 903 983 974 985 942 986 984 979 986 987 985 963 988 978 939 980 914 970 990 989 994 954 981 995 993 996 975 997 999 999 997
0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1
1006 1994
1000 1999
1000 1992
1004 1999
1003 1992
