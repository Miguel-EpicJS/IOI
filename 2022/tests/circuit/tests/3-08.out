4faa78c22ee35083e49b65601698a1bd98a87a0f
OK
230025634
903738266
10382050
690967620
797613426
